### PR TITLE
D Trades Questionnaire

### PR DESCRIPTION
**Pull request by bitdivision**
### What is the context of this PR?
[trello](https://trello.com/c/UFsAJ6ab/2569-d-trades-mts-build-for-testing)

Add the initial D-Trades questionnaire from author, and make minor changes that weren't available in author.

Schema name is mts_0001.json

### How to review 
Check against the slides from trello.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: data/en/d_trades.json [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+                                        }
+                                    ]
+                                },
+                                "description": "",
+                                "type": "General",
+                                "answers": [{
+                                    "id": "answer3821",
+                                    "mandatory": true,
+                                    "type": "Number",
+                                    "label": "Approximate  percentage from non-retail activities",
```
Comment (MebinAbraham) [inline data/en/d_trades.json]: Double spaces

File: data/en/d_trades.json [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+                                    "description": "",
+                                    "guidance": {
+                                        "show_guidance": "Show examples of commentary on changes to retail sales",
+                                        "hide_guidance": "Hide examples of commentary on changes to retail sales",
+                                        "content": [{
+                                            "title": "\u2018In-store promotion\u2019",
+                                            "description": "\u201cOffer on wine for the whole month. The promotion was available in-store and online, contributing to an increase in both in-store and online sales.\u201d"
+                                        }, {
+                                            "title": "\u2018Special events (for example, sporting events)\u2019",
+                                            "description": "\u201cThis was the month before the start of  World Cup 2018 (football), we recorded an increase in sales of audio-visual equipment (for example, televisions and audio equipment). This led to an increase in sales both in-store and online.\u201d"
```
Comment (MebinAbraham) [inline data/en/d_trades.json]: Double space

File: data/en/d_trades.json [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+                                    }
+                                }]
+                            }]
+                        },
+                        {
+                            "id": "block2939",
+                            "type": "Question",
+                            "questions": [{
+                                "id": "question2939",
+                                "title": "For the period {{ metadata['ref_p_start_date'] | format_date }} to {{ metadata['ref_p_end_date'] | format_date }}, did {{ metadata['trad_as_or_ru_name'] }} have turnover from <em>any activities other than retail sales</em>?   ",
```
Comment (MebinAbraham) [inline data/en/d_trades.json]: Unneeded spaces before `?`. Author -_-

File: data/en/d_trades.json [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+                                    }
+                                }
+                            ]
+                        },
+                        {
+                            "id": "block2940",
+                            "type": "Question",
+                            "questions": [{
+                                "id": "question2940",
+                                "title": "For the period {{ format_conditional_date (answers['period-from-answer'], metadata['ref_p_start_date'])}} to {{ format_conditional_date (answers['period-to-answer'], metadata['ref_p_end_date'])}}, approximately what <em>percentage </em>of {{ metadata['trad_as_or_ru_name'] }} turnover was  <em>from other activities</em>?   ",
```
Comment (MebinAbraham) [inline data/en/d_trades.json]: Unneeded spaced before `?`

File: data/en/d_trades.json [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+                                    }
+                                }
+                            }]
+                        },
+                        {
+                            "id": "retail-sales-block",
+                            "type": "Question",
+                            "questions": [{
+                                "id": "question2934",
+                                "title": "For the period {{ format_conditional_date (answers['period-from-answer'], metadata['ref_p_start_date'])}} to {{ format_conditional_date (answers['period-to-answer'], metadata['ref_p_end_date'])}}, how much were your <em>retail sales including VAT</em>? ",
```
Comment (MebinAbraham) [inline data/en/d_trades.json]: Unneeded space before `?`

File: data/en/d_trades.json [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+                            "skip_conditions": [{
+                                "when": [{
+                                    "id": "retail-sales-answer",
+                                    "condition": "equals",
+                                    "value": 0
+                                }]
+                            }],
+                            "questions": [{
+                                "id": "question2935",
+                                "title": "Of the {{ format_currency(answers['retail-sales-answer'], 'GBP') }} retail sales, how much were <em>online sales</em>? ",
```
Comment (MebinAbraham) [inline data/en/d_trades.json]: Unneeded space before `?`

**Review by jonnyshaw89 — DISMISSED** [review]

File: data/en/mts_0001.json [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+                                        "label": "Yes this is correct",
+                                        "value": "Yes"
+                                    }, {
+                                        "label": "No I need to change this",
+                                        "value": "No"
+                                    }],
+                                    "mandatory": true
+                                }],
+                                "id": "confirm-zero-retail-sales-question",
+                                "title": "For the period {{ format_conditional_date (answers['period-from'], metadata['ref_p_start_date'])}} to {{ format_conditional_date (answers['period-to'], metadata['ref_p_end_date'])}}, the value of retail sales was {{ format_currency(answers['retail-sales-answer'], 'GBP') }}, is this correct?"
```
Comment (MebinAbraham) [inline data/en/mts_0001.json]: This filter is using wrong arguments. Hence its displaying wrong dates It should be used `period-from-answer` and `period-to-answer`

**Review by MebinAbraham — CHANGES_REQUESTED** [review]

File: data/en/mts_0001.json (hunk 7 above)
Comment (MebinAbraham) [reply data/en/mts_0001.json]: Other places where we use the metadata date, are we supposed to be using the user inputted date?. Like here?

https://github.com/ONSdigital/eq-survey-runner/blob/1d9e309377f7c41f3c256526cf9749679fa671ed/data/en/mts_0001.json#L523

Comment (bitdivision) [reply data/en/mts_0001.json]: Correct on both counts. Good spots.

**Comment by bitdivision** [comment]
Donna has looked through and found some issues, which I've corrected. She is happy with it as is.

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
Routing missing?
According to slides: ***On retail sales including VAT:***
Expected: If 0 then route to ‘Percentage turnover from other activities?
Actual: Next block

**Comment by bitdivision** [comment]
As noted in Donna's comments, the confirmation question should always be shown for both questions. So that routing is redundant.

**Comment by MebinAbraham** [comment]
Sildes need to be updated then i guess.

**Review by MebinAbraham — DISMISSED** [review]

**Comment by bitdivision** [comment]
@MebinAbraham 	@jonnyshaw89 	Note that I've update the lock file to resolve security issue with requests (since it broke travis). Since I was there, I also removed the pinned versions of moto and boto that we had in the Pipfile. The issue they had with conflicting versions of boto-core has now been resolved.

**Comment by bitdivision** [comment]
@jonnyshaw89 	@MebinAbraham 	Can you rereview? I had to make quite a few code changes to fix the linter, so please make sure I haven't done anything stupid.

The setup.cfg rules for flake8 were being overridden by tox.ini, so I've removed those.

**Review by jonnyshaw89 — APPROVED** [review]
👍

File: data/en/mts_0001.json [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+                            "description": "You can select the dates of the period you are reporting for, if the given dates are not appropriate."
+                        }],
+                        "questions": [{
+                            "question": "Retail Sales including VAT",
+                            "content": [{
+                                "description": "Include:",
+                                "list": ["VAT", "online sales"]
+                            }, {
+                                "description": "Exclude:",
+                                "list": ["sales from catering facilities used by customers (for example, in-store cafes)", "revenue from mobile phone network commission and top-up", "NHS receipts", "lottery sales and commission from lottery sales", "sales of car accessories and motor vehicles"]
```
Comment (dwyeradam) [inline data/en/mts_0001.json:74]: My OCD is annoyed that the order of these is wrong. I won't block it on this but it hurts.

**Review by dwyeradam — APPROVED** [review]
Can't approve myself as the slides are incorrect. This approval is therefore the proxy for Donna.

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**